tests/test_ble_interface_core.py

```
@@ -388,29 +453,1263 @@ def test_state_manager_closing_only_for_disconnect() -> None:
 
-    Parameters
-    ----------
-    monkeypatch : pytest.MonkeyPatch
-    """
+    monkeypatch.setattr(iface, "_validate_connection_preconditions", lambda: None)
+    monkeypatch.setattr(iface, "_get_existing_client_if_valid", lambda _req: None)
+    monkeypatch.setattr(iface, "_raise_if_duplicate_connect", lambda _key: None)
+
+    with pytest.raises(
+        BLEInterface.BLEError,
+        match="invalid connect_timeout: connect_timeout must be a finite positive number of seconds.",
```

Ruff (0.15.4)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)